src/tss2-tcti/tctildr.c

```diff
@@ -85,3 +97,365 @@ tcti_from_info(TSS2_TCTI_INFO_FUNC infof,
 
     return TSS2_RC_SUCCESS;
 }
+/*
+ * name_conf in the form "tcti-name:tcti-conf"
+ * copies 'tcti-name' component to 'name' buffer
+ * copies 'tcti-conf' component to 'conf' buffer
+ * handled name_conf forms:
+ * - "", ":" -> both name and conf are left unchanged
+ * - "tcti-name", "tcti-name:" -> tcti-name copied to 'name', 'conf'
+ *   unchanged
+ * - ":tcti-conf" -> tcti-conf copied to 'conf', 'name' unchanged
+ * - "tcti-name:tcti-conf" - "tcti-name" copied to 'name,', "tcti-conf"
+ *   copied to 'conf'
+ */
+TSS2_RC
+tctildr_conf_parse (const char *name_conf,
+                    char *name,
+                    char *conf)
```

Comment:
My only concern is that since `tctildr_conf_parse` requires the caller to pass already allocated `name` and `conf`, we can't stop them from allocating something which would be to small, we can tell them to, but they may not listen. So I think that this is an appropriate place to add `strn`-ish arguments where we make the caller specify `name_length` and `conf_length`. Granted, this isn't an exported function, but could save some maintainer years down the road some headache if they try to pass in buffers that are too small.

Reply:
But this is very take it or leave it, since its an internal function

Reply:
I had a version that took the length but I went the route of less code, more assumptions. Since it's an internal function (as you point out) I figured the assumptions were reasonable. Would prefer to "leave it" as is.